packages/wonder-blocks-modal/src/components/flexible-dialog.tsx

```diff
@@ -0,0 +1,225 @@
+    /**
+     * When set, provides a component that can render content above the top of the modal;
+     * when not set, no additional content is shown above the modal.
+     * This prop is passed down to the ModalDialog.
+     */
+    above?: React.ReactNode;
+    /**
+     * When set, provides a component that will render content below the bottom of the modal;
+     * when not set, no additional content is shown below the modal.
+     * This prop is passed down to the ModalDialog.
+     *
+     * NOTE: Devs can customize this content by rendering the component assigned to this prop with custom styles,
+     * such as by wrapping it in a View.
+     */
+    below?: React.ReactNode;
+    /**
+     * When set, overrides the default role value. Default role is "dialog"
+     * Roles other than dialog and alertdialog aren't appropriate for this
+     * component
+     */
+    role?: "dialog" | "alertdialog";
+    /**
+     * Optional custom styles.
+     */
+    style?: StyleType;
+    /**
+     * Test ID used for e2e testing. This ID will be passed down to the Dialog.
+     */
+    testId?: string;
+    /**
+     * The ID of the content describing this dialog, if applicable.
+     */
+    "aria-describedby"?: string;
+};
+
+type RenderProps = {
+    title: React.ReactNode | string;
+};
+/**
+ * A more flexible modal variant with fewer layout constraints. It can receive
+ * a custom background (image or color), a title for the main heading, and that
+ * title can optionally render in the content area through a render prop.
+ *
+ * One of the following is required for labeling the dialog:
+ * - title content (React node or string)
+ * - aria-label (string)
+ * - aria-labelledby (string ID reference)
+ *
+ * ### Usage
+ *
+ * ```jsx
+ * import {FlexibleDialog} from "@khanacademy/wonder-blocks-modal";
+ * import {BodyText} from "@khanacademy/wonder-blocks-typography";
+ *
+ * <FlexibleDialog
+ *     title={<Heading size="xxlarge" id="main-heading">Select mission</Heading>}
+ *     content={
+ *         <BodyText>
+ *             {`Lorem ipsum dolor sit amet, consectetur adipiscing
+ *             elit, sed do eiusmod tempor incididunt ut labore et
+ *             dolore magna aliqua. Ut enim ad minim veniam,
+ *             quis nostrud exercitation ullamco laboris nisi ut
+ *             aliquip ex ea commodo consequat. Duis aute irure
+ *             dolor in reprehenderit in voluptate velit esse
+ *             cillum dolore eu fugiat nulla pariatur. Excepteur
+ *             sint occaecat cupidatat non proident, sunt in culpa
+ *             qui officia deserunt mollit anim id est.`}
+ *         </BodyText>
+ *     }
+ * />
+ * ```
+ */
+const FlexibleDialog = ({
+    onClose,
+    title,
+    content,
+    above,
+    below,
+    style,
+    closeButtonVisible = true,
+    testId,
+    titleId,
+    role,
+    backgroundStyles,
+    ...accessibilityProps
+}: Props): React.ReactElement => {
+    return (
+        <Id id={titleId}>
```

Comment:
Since `FlexibleDialog` is a functional component already, can we use the `React.useId` hook? 😄